nixos/modules/config/stub-ld.nix

```
@@ -0,0 +1,75 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) last splitString mkOption types mdDoc mkIf optionals;
+
+  message = ''
+    You seem to be trying to run a dynamically linked executable that
+    wasn't made for NixOS. This is possible, but not without extra steps,
+    because NixOS, by design, does not have a global library path.
+
+    There are several solutions available:
+
+    1. If the program is packaged in nixpkgs, use that version.
+
+    2. Write a nix expression for the program to package it
+       yourself. There are 3 approaches to this:
```

```suggestion
       yourself. There are multiple approaches to this:
```

Don't hard code list length, this will just entail maintenance overhead in the future.

---

:laughing: true.